File: src/main/java/ca/yyx/hu/aap/AapSsl.java
package ca.yyx.hu.aap;

/**
 * @author algavris
 * @date 14/02/2017.
 */
interface AapSsl {
    int prepare();
    void handshake();
    ByteArray bioRead();
    int bioWrite(int start, int length, byte[] buffer);
    ByteArray decrypt(int start, int length, byte[] buffer);
    ByteArray encrypt(int offset, int length, byte[] buffer);
}

File: src/main/java/ca/yyx/hu/aap/AapSsl.kt
package ca.yyx.hu.aap

/**
 * @author algavris
 * *
 * @date 14/02/2017.
 */
internal interface AapSsl {
    fun prepare(): Int
    fun handshake()
    fun bioRead(): ByteArrayWithLimit?
    fun bioWrite(start: Int, length: Int, buffer: ByteArray): Int
    fun decrypt(start: Int, length: Int, buffer: ByteArray): ByteArrayWithLimit?
    fun encrypt(offset: Int, length: Int, buffer: ByteArray): ByteArrayWithLimit?
}
